pyproject.toml

```diff
@@ -1,4 +1,13 @@
 [build-system]
-requires = ["setuptools", "setuptools_scm[toml]>=6.2",
-            "packaging", "jinja2>=2.10.3", "numpy>=1.25,<2"]
+requires = [
+    "setuptools",
+    "setuptools_scm>=6.2",
+    "jinja2>=2.10.3",
+    "numpy>=1.25,<2; python_version<='3.11'",
```

Comment:
Should we use <3.12 to be the exact opposite of >=3.12?

Reply:
Wondering how 3.11.3 evaluates for example, is it <=3.11?

Reply:
Will that pick up 3.12rc? I can never tell how these operators work.

Reply:
This is inferred from https://github.com/scipy/scipy/blob/main/pyproject.toml

Reply:
Well, I made the change you requested and CI is still green, so I guess it is okay after all. Please re-review. Thanks!